### PR TITLE
Fixed sandbox-related hang

### PR DESCRIPTION
**Pull request by TimOliver**
This is a fix to the issue of the Browser hanging when a Realm document is closed.

There's a bit of an incompatibility with the way OS X sandboxing and Realm behave together. When a Realm file is opened, the sandbox rights to the folder hosting the Realm file (So Realm has write permission to create new files around the original .realm file) are requested and once approved by the user, opened. That reference is held onto for the duration of that .realm file being open.

When the document is closed, the sandbox handle is closed in the document's `dealloc` method. Unfortunately, the RLMRealm object itself isn't released at that point, but it does end up getting released automatically a little bit later. The problem we're experiencing here is that in Realm's internal C++ destructor method, it tries to delete certain files on the disk (Probably just as a last second cleanup), and as this is occurring after the sandbox is closed, it causes the destructor method to fail, and throw an exception.

While the easiest solution might be to simply 'ignore' closing sandbox handles once a document is opened, apparently OS X keeps track of how many documents a single app has open, and has an internal limit to the number of active sandbox handles it has. The ideal solution here would be to somehow get Realm Browser to explicitly and immediately release the RLMRealm object in the RLMDocument's dealloc method, but I'm still exploring if it's possible to re-architect the logic to allow that.

For the time being, while not ideal, this solution works. As the RLMDocument object is released, a block is scheduled to close the sandbox handle a minute later, much later after Realm has finished cleaning itself.

Down the line, I plan to explore how this can be done much more elegantly. It might make more sense to detach the sandbox logic from RLMDocument and have it as a globally managed state in the app.

### CODE REVIEW

File: RealmBrowser/Classes/RLMDocument.m [hunk 1]
```diff
@@ -131,12 +131,17 @@ - (id)initForURL:(NSURL *)urlOrNil withContentsOfURL:(NSURL *)contentsURL ofType
...
 {
     [self.presentedRealm.realm removeNotification:self.changeNotificationToken];
-    self.presentedRealm = nil;
     
-    //Release the sandbox rights on the next run-loop iteration
-    //as there are periodically some file operations in RLMRealm's destructor method
-    dispatch_sync(dispatch_get_main_queue(), ^{
-        [self.securityScopedURL stopAccessingSecurityScopedResource];
+    //In certain instances, RLMRealm's C++ destructor method will attempt to clean up
+    //specific auxillary files belonging to this realm file.
```
Comment (jpsim) [inline RealmBrowser/Classes/RLMDocument.m]: auxiliary

File: RealmBrowser/Classes/RLMDocument.m [hunk 2]
```diff
@@ -131,12 +131,17 @@ - (id)initForURL:(NSURL *)urlOrNil withContentsOfURL:(NSURL *)contentsURL ofType
...
+    //If the destructor call occurs after the sandbox scope has been released here, and it attempts to delete
+    //any files, RLMRealm will throw an exception.
+    //Mac OS X apps only have a finite number of sandbox handles allocated at once, so while it's not necessary
+    //to release the sandbox scope straight away, it is still necessary. As such, this will release the handle a minute
```
Comment (jpsim) [inline RealmBrowser/Classes/RLMDocument.m]: > so while it's not necessary to release the sandbox scope straight away, it is still necessary.

lolwut? Do you mean "it is still necessary to do so *eventually*"?

Comment (TimOliver) [inline RealmBrowser/Classes/RLMDocument.m]: There's an upper limit to the number of sandbox handles an app can have open at once. It's a really high number (So we could in theory get away with not even bothering to close them), but it'd still be cleaner to close them.

Comment (jpsim) [inline RealmBrowser/Classes/RLMDocument.m]: That I got, but the sentence in the comment doesn't make sense as-is. Consider rewording to what I proposed.

**Pull request merged**